structs.go

```diff
@@ -2467,7 +2509,7 @@ type Entitlement struct {
 
 	// The ID of the user that is granted access to the entitlement's sku
 	// Only available for user subscriptions.
-	UserID string `json:"user_id,omitempty"`
+	UserID *string `json:"user_id,omitempty"`
```

Comment:
Is there ever a point at which we'd expect/want a default string value here? Wondering if the introduction of pointer risk is necessary or if we can roll with default values without losing any important data (no user has no ID, no guild has no ID, no entitlement has a start time at zero, etc.)

Reply:
You are right, thank you. However, I would keep all the optional timestamps as pointers, since this is also done in other structs (e.g., Channel, Member, Invite).